app/src/main/java/bacon/john/bitcoinpricewatch/CurrencyDetailFragment.java

```diff
@@ -0,0 +1,142 @@
+        super.onCreate(savedInstanceState);
+
+        if (getArguments().containsKey(CURRENCY_CODE)) {
+            // Load the dummy content specified by the fragment
+            // arguments. In a real-world scenario, use a Loader
+            // to load content from a content provider.
+            String currencyCode = getArguments().getString(CURRENCY_CODE);
+            mCurrency = BitcoinCurrencyManager.sharedInstance().getCachedCurrency(currencyCode);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_currency_detail, container, false);
+        return rootView;
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        this.refreshView();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        long delay = 10000;
+        long timeSinceLastUpdate = System.currentTimeMillis() - mCurrency.getUpdatedAt();
+
+        // refresh immediately if it has been more then 10 seconds
+        if (timeSinceLastUpdate >= 10000) {
+            refresh();
+        } else {
+            delay = 10000 - timeSinceLastUpdate;
+        }
+
+        mTimer = new Timer();
+        mTimer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                refresh();
+            }
+        }, delay, 10000);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mTimer.cancel();
+        mTimer = null;
+    }
+
+    private void refresh() {
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                BitcoinCurrencyManager.sharedInstance().getRemoteCurrencyPrices(mCurrency, new BitcoinCurrencyManager.CurrencyCallback() {
+                    @Override
+                    public void callback(BitcoinCurrencyModel currency, Exception error) {
+                        if (getActivity() != null) {
+                            if (error == null) {
+                                mCurrency = currency;
+                                refreshView();
+                            } else {
+                                if (getView() != null) {
+                                    ((TextView)getView().findViewById(R.id.currency_detail)).setText("There was a problem loading data for this currency - please try again later.");
+                                }
+                            }
+                        }
+                    }
+                });
+            }
+        });
+    }
+
+    private void refreshView() {
+        if (getView() != null) {
+            if (mCurrency.getUpdatedAt() > 0) {
+                String displayString = "Ask Price:\n" + mCurrency.getAsk() + "\n\n";
```

Comment:
wish I had time to make this more beautiful, but for now it's just a string with a bunch of newlines